lib/internal/streams/readable.js

```diff
@@ -251,9 +256,17 @@ function readableAddChunk(stream, chunk, encoding, addToFront) {
         if (addToFront && state.encoding) {
           // When unshifting, if state.encoding is set, we have to save
           // the string in the BufferList with the state encoding.
+
           chunk = Buffer.from(chunk, encoding).toString(state.encoding);
         } else {
-          chunk = Buffer.from(chunk, encoding);
+          const enc = normalizeEncoding(encoding);
+
+          if (enc === 'utf8') {
+            const buf = encoder.encode(chunk);
+            chunk = new FastBuffer(buf.buffer, buf.byteOffset, buf.byteLength);
```

Comment:
```suggestion
            chunk = new FastBuffer(
              TypedArrayPrototypeGetBuffer(buf),
              TypedArrayPrototypeGetByteOffset(buf),
              TypedArrayPrototypeGetByteLength(buf),
            );
```

Reply:
Thanks for the review. I updated my pull request.

Reply:
Let's run another benchmark to ensure it doesn't affect performance (sometimes using primordials have weird effects on V8 optimizations)

Reply:
Good to know. I'm running it.